external/googleapi/g_utils.py

```diff
@@ -11,3 +16,52 @@ def parse_lat_lng(geo_result_obj):
 
     # default case if there was no result returned
     return (None, None)
+
+
+def normalize_us_address(address) -> Optional[Address]:
+    if not address:
+        return None
+
+    response_list = fetch_geocode(address, region="us")
+
+    if not response_list:
+        return None
+
+    response = response_list[0]
+
+    state = None
+    city = None
+    route = ""
+    street_num = ""
+
+    for comp in response.get("address_components"):
+        types = comp.get("types")
+
+        if not types:
+            continue
+
+        if "administrative_area_level_1" in types:
+            state = comp.get("short_name")
+
+        if "street_number" in types:
+            street_num = comp.get("long_name")
+
+        if "route" in types:
+            route = comp.get("long_name")
+
+        # https://stackoverflow.com/a/49640066/1556838
+        if "locality" in types or "sublocality_level_1" in types:
+            city = comp.get("long_name")
+
+    zip_code = response.get("postal")
+    loc = response.get("geometry").get("location")
+    lat, lon = loc["lat"], loc["lng"]
+
+    return Address(
+        street=" ".join(filter(lambda x: x, [street_num, route])),
```

Comment:
Would you mind explaining what this line is doing?

Reply:
`filter` filters falsy values in the `[street_num, route]`. In this case, empty string or None are expected to be filtered.

`join` joins `List[str]` with `" "`.

For instance,
street_num = "6", route = "Jay St" => "6 Jay St"
street_num = None, route = "Jay St" => "Jay St"
street_num = None, route = None => ""
street_num = "6", route = None => "6"